tensorflow/core/kernels/maxpooling_op_sycl.h

```diff
@@ -45,28 +45,28 @@ template <typename T>
 class MaxPool2DSYCL {
  public:
   static constexpr auto read_mode = cl::sycl::access::mode::read;
-  static constexpr auto write_mode = cl::sycl::access::mode::write;
+  static constexpr auto write_mode = cl::sycl::access::mode::discard_write;
   static constexpr auto global_access = cl::sycl::access::target::global_buffer;
   using read_accessor =
       cl::sycl::accessor<uint8_t, 1, read_mode, global_access>;
   using write_accessor =
       cl::sycl::accessor<uint8_t, 1, write_mode, global_access>;
 
-  MaxPool2DSYCL(const int n_threads, const PoolParameters& params,
+  MaxPool2DSYCL(const int n_threads, const int output_size,
+                const PoolParameters& params,
                 const read_accessor input_accessor,
                 write_accessor output_accessor)
       : n_threads_{n_threads},
```

Comment:
n_threads_ has not been used in this kernel.

Reply:
It was already there so I kept it. Should I remove that in this PR?

Reply:
yes